src/Tools/dotnet-monitor/Extensibility/ProgramExtension.cs

```diff
@@ -90,6 +106,7 @@ public async Task<EgressArtifactResult> EgressArtifact(ExtensionEgressPayload co
                 UseShellExecute = false,
             };
             pStart.ArgumentList.Add(ExtensionTypes.Egress);
+            _processEnvironmentVariables.ToList().ForEach(x => pStart.EnvironmentVariables[x.Key] = x.Value);
```

Comment:
```suggestion
            foreach ((string key, string value) in _processEnvironmentVariables)
            {
                _pStart.Environment.Add(key, value);
            }
```

Or write a `DictionaryExtensions.AddRange<TKey, TValue>(this IDictionary<TKey, TValue> target, IDictionary<TKey, TValue> source)` extension method that does this and call it here if you think this is reusable.